client/app/services/navigateTo.js

```diff
@@ -1,8 +1,15 @@
 import { isString } from 'lodash';
-import { $location, $rootScope } from '@/services/ng';
+import { $location, $rootScope, $route } from '@/services/ng';
 
-export default function navigateTo(url, replace = false) {
+export default function navigateTo(url, replace = false, reload = true) {
   if (isString(url)) {
+    if (!reload) {
+      const lastRoute = $route.current;
+      const un = $rootScope.$on('$locationChangeSuccess', () => {
+        $route.current = lastRoute;
+        un();
+      });
+    }
```

Comment:
@arikfr and @kravets-levko, this was the logic I had to introduce to allow changing the URL without reloading. Technically when you update the path without reloading you're not actually navigating anywhere, but this was the method used to change angular route on React. LMK if you think it's better to create a new method and move this or rename it.

BTW, I thought about using a JS method so we'd not depend on Angular:
```javascript
// change url without reloading
if (window.history.pushState) {
  window.history.pushState({}, '', url);
}
```
The problem is that Angular didn't work well with it (a lot of `$digest`'s in sequence). Anyway, this will be replaced when we move to another router.

Reply:
😖 

@kravets-levko any thoughts or we can merge this?

Reply:
> BTW, I thought about using a JS method so we'd not depend on Angular:

It's possible, but it may break some AngularJS internals ($location may become out of sync with url). Also it may be a similar problem when we'll move to some React router.

@arikfr I think we can merge this. @gabrieldutra maybe just add a short comment which describes that patch and note that we should keep this in mind when replacing Angular's router with new one.

Reply:
@kravets-levko I added a comment with the well known `ANGULAR_REMOVE_ME` note :+1:.